File: learn_algorithm/leetcode/src/main/java/com/gmy/leetcode/link/ListNode.java
package com.gmy.leetcode.link;

public class ListNode {

    public int val;
    public ListNode next;

    public ListNode() {}

    public ListNode(int val) { this.val = val; }
    
    public ListNode(int val, ListNode next) { this.val = val; this.next = next; }

    public ListNode(int... val) {
        ListNode build = build(val);
        this.val = build.val;
        this.next = build.next;
    }
    
    public ListNode setNext(int val) {
        ListNode listNode = new ListNode(val);
        this.next = listNode;
        return this.next;
    }

    public static ListNode build(int[] arr) {
        if (arr == null) {
            return null;
        }
        ListNode listNodes = new ListNode(arr[0]);
        ListNode head = listNodes;
        for (int i = 1; i < arr.length; i++) {
            listNodes.setNext(arr[i]);
            listNodes = listNodes.next;
        }
        return head;
    }

    @Override
    public String toString() {
        String result =  "{" +
                "\"val\":" + val;
        if (next != null) {
            result += ", \"next\":" + next.toString();
        }
        result += "}";
        return result;
    }
}

File: python/labuladong/list_node.py
# -*- coding: utf-8 -*-

"""
ListNode 公共类
"""


class ListNode(object):
    pass


class ListNode(object):

    def __init__(self, val: int):
        self.val = val
        self.next = None

    def set_next(self, val: int) -> ListNode:
        result = ListNode(val)
        self.next = result
        return result

    def __str__(self) -> str:
        result = "{\"val\":" + self.val.__str__()
        if self.next is not None:
            result += ", \"next\":" + self.next.__str__()
        result += "}"
        return result

    def get_next(self) -> ListNode:
        return self.next


def main():
    node = ListNode(1)
    node.set_next(2).set_next(3)
    print(node)


if __name__ == '__main__':
    main()


